db/system_keyspace.cc

```diff
@@ -2162,12 +2162,46 @@ future<> system_keyspace::update_peer_info(gms::inet_address ep, locator::host_i
     slogger.debug("{}: values={}", query, values);
 
     co_await _qp.execute_internal(query, db::consistency_level::ONE, values, cql3::query_processor::cache_internal::yes);
+
+    if (auto* cache = get_peers_cache()) {
+        cache->id_to_ip[hid] = ep;
+        cache->ip_to_id[ep] = hid;
+    }
```

Comment:
Should we be concerned about a case where `execute_internal` fails (e.g. timeout)? This might leave the cache in an inconsistent state for a short while.

Maybe we should consider invalidating the cache on error here.

Reply:
If execute_internal fails, then we don't know if the mutations were applied or not.